datafusion/physical-plan/src/aggregates/group_values/row.rs

```diff
@@ -111,45 +123,90 @@ impl GroupValues for GroupValuesRows {
             None => self.row_converter.empty_rows(0, 0),
         };
 
+        let mut store_gp_hashes = match self.group_hashes.take() {
+            Some(group_hashes) => group_hashes,
+            None => vec![],
+        };
+
         // tracks to which group each of the input rows belongs
         groups.clear();
 
-        // 1.1 Calculate the group keys for the group values
-        let batch_hashes = &mut self.hashes_buffer;
-        batch_hashes.clear();
-        batch_hashes.resize(n_rows, 0);
-        create_hashes(cols, &self.random_state, batch_hashes)?;
-
-        for (row, &hash) in batch_hashes.iter().enumerate() {
-            let entry = self.map.get_mut(hash, |(_hash, group_idx)| {
-                // verify that a group that we are inserting with hash is
-                // actually the same key value as the group in
-                // existing_idx  (aka group_values @ row)
-                group_rows.row(row) == group_values.row(*group_idx)
-            });
-
-            let group_idx = match entry {
-                // Existing group_index for this group value
-                Some((_hash, group_idx)) => *group_idx,
-                //  1.2 Need to create new entry for the group
-                None => {
-                    // Add new entry to aggr_state and save newly created index
-                    let group_idx = group_values.num_rows();
-                    group_values.push(group_rows.row(row));
-
-                    // for hasher function, use precomputed hash value
-                    self.map.insert_accounted(
-                        (hash, group_idx),
-                        |(hash, _group_index)| *hash,
-                        &mut self.map_size,
-                    );
-                    group_idx
-                }
-            };
-            groups.push(group_idx);
+        // let hash_values: Option<&ArrayRef> = None;
```

Comment:
The most surprised things is I got the similar result as main if I uncomment this line...

The only difference is one iterate U64Array, another calculate the hashes and get `Vec<u64>` directly